[Darylgolden] iP

Darylgolden opened the pull request:
### GoldBot
> "This is an important quote" - Me

GoldBot helps you take charge of your life by keeping track of your tasks. It's,

- CLI-based
- easy to use

All you need to do is

1. Download it from [here](https://github.com/Darylgolden/ip/releases/tag/v0.1.0)
2. Run it! 😺 

Features:

- [ ] Managing tasks
- [ ] Managing deadlines and events
- [ ] Task search

Here's the main method:
```java
public class GoldBot {
    public static void main(String[] args) {
        System.out.println("test");
        Storage storage = new Storage("./data/data.txt");
        TaskList taskList;
        try {
            taskList = storage.createTaskList();
        } catch (IOException e) {
            System.out.println("Unexpected error occured.");
            return;
        } catch (FileCorruptionException e) {
            System.out.println("File corrupted.");
            return;
        }
        ChatSession session = new ChatSession(taskList);
        session.initChat();
        
        Scanner scanner = new Scanner(System.in);

        while (session.shouldContinueSession) {
            String input = scanner.nextLine();
            session.handleMessage(input);
        }
        scanner.close();
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Commit:
docs/README.md: Tweak document template

Let's tweak the docs/README.md (which is used as the user guide)
to fit Duke better. Specifically,

1. mention product name in the title
2. mention adding a product screenshot and a product intro
3. tweak the flow to describe feature-by-feature

Darylgolden committed:
created command interface and chatsession class, renamed duke

Darylgolden committed:
level 0

Darylgolden committed:
create task class

Darylgolden committed:
refactored logic into chatsession

Darylgolden committed:
start bye command and impl of termination commands

Darylgolden committed:
create printMessage method in ChatSession

Darylgolden committed:
method for printing messages and for terminating sesh

Darylgolden committed:
level 1

Darylgolden committed:
level 2

Darylgolden committed:
level 3

Darylgolden committed:
add todo

Darylgolden committed:
create event, deadline, subcommand parsing util

Darylgolden committed:
add logic to print dates

Darylgolden committed:
implement deadline parser

Darylgolden committed:
level 4

Darylgolden committed:
Add support for semiautomated testing

Darylgolden committed:
Add basic exceptions

Darylgolden committed:
Added exception handling

Darylgolden committed:
Added delete functionality

Darylgolden committed:
Added data export functionality to Task

Darylgolden committed:
Added export for Deadline, Event, Task

Darylgolden committed:
Create logic for initialization

Darylgolden committed:
Finish level 7

Darylgolden committed:
Merge branch 'branch-level-7'

Darylgolden committed:
Switched deadline to use Java DateTime

Darylgolden committed:
Changed Event to use java LocalDateTime

Darylgolden committed:
Finished DateTime integration

Darylgolden committed:
Merge branch 'branch-Level-8'

Darylgolden committed:
Refactored logic into Ui and Parser

Darylgolden committed:
Merge branch 'A-MoreOOP'

Darylgolden committed:
Refactored classes into packages

Darylgolden committed:
Merge branch 'A-Packages'

Darylgolden committed:
Merge branch 'add-gradle-support'

Darylgolden committed:
Configure gradle

Darylgolden committed:
Add test for subcommand parser

Darylgolden committed:
Add test for Event

Darylgolden committed:
Merge branch 'A-JUnit'

Darylgolden committed:
Add manifest

Darylgolden committed:
Add documentation

Darylgolden committed:
Rewrite code to match coding standard

Darylgolden committed:
Implement Find command

Darylgolden committed:
Merge branch 'A-JavaDoc'

Darylgolden committed:
Merge branch 'A-CodingStandard'

Darylgolden committed:
Merge branch 'Level-9'

sindhurajain commented on src/main/java/duke/ChatSession.java:
```diff
@@ -0,0 +1,35 @@
+    public boolean continueSession;
+
+    ChatSession(TaskList taskList) {
+        this.continueSession = true;
```
Shouldn't boolean variables be named to sound like booleans? Eg isSessionContinued

sindhurajain commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,68 @@
+            new ToDoCommand(), 
+            new EventCommand(), 
+            new DeadlineCommand(),
+            new Delete()
```
Perhaps you should use consistent names - eg DeleteCommand, MarkCommand, etc

sindhurajain commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,84 @@
+        default:
+            throw new Exception();
+        }
+        return t;
```
Good job on leaving no indentation for case clauses :+1:

sindhurajain commented on src/main/java/duke/SubcommandParser.java:
```diff
@@ -0,0 +1,44 @@
+        String[] words = text.split(" ");
+        List<String> initial = new ArrayList<>();
+        Pattern p = Pattern.compile("\\" + initializer + "\\w*");
+        int i = 0;
```
The variable i has a large scope, so perhaps give it a longer, more intuitive name

sindhurajain commented on src/main/java/duke/SubcommandParser.java:
```diff
@@ -0,0 +1,44 @@
+        List<String> initial = new ArrayList<>();
+        Pattern p = Pattern.compile("\\" + initializer + "\\w*");
+        int i = 0;
+        boolean matched = false;
```
Perhaps change this to isMatched as boolean variables should be named to sound like booleans

sindhurajain commented on src/main/java/duke/SubcommandParser.java:
```diff
@@ -0,0 +1,44 @@
+        while (i < words.length) {
+            String word = words[i];
+            i++;
+            Matcher m = p.matcher(word);
```
Maybe you can include more detail in the naming of the Matcher variable. Perhaps patternMatcher or patternToWordMatcher?

sindhurajain commented on src/main/java/duke/commands/DeadlineCommand.java:
```diff
@@ -0,0 +1,33 @@
+                default:
+                    // add exception handling later
+                    break;
+            }
```
Shouldn’t there be no indentation for case clauses?

sindhurajain commented on src/main/java/duke/commands/Delete.java:
```diff
@@ -0,0 +1,14 @@
+
+
+public class Delete implements NamedCommand {
+    public String getName() { return "delete"; }
```
Shouldn't the code in the curly braces be on the next line?

sindhurajain commented on src/main/java/duke/commands/EventCommand.java:
```diff
@@ -0,0 +1,36 @@
+                default:
+                    // add exception handling later
+                    break;
+            }
```
Shouldn’t there be no indentation for case clauses?

sindhurajain commented on src/main/java/duke/commands/ListTasks.java:
```diff
@@ -0,0 +1,11 @@
+    public void execute(ChatSession session, String commandArgs) {
+        session.printMessage("Here are the tasks in your list:"
+        + System.lineSeparator()
+        + session.taskList.generateName());
```
Good job on keeping the line length short

sindhurajain commented on src/main/java/duke/commands/Mark.java:
```diff
@@ -0,0 +1,13 @@
+import duke.tasks.Task;
+
+public class Mark implements NamedCommand {
+    public String getName() { return "mark"; }
```
Shouldn't code in the curly braces be on the next line?

sindhurajain commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,64 @@
+        this.dataFile = dataFile;
+    }
+
+    public String generateName() {
```
Perhaps use verbs for method names as per the code quality guidelines? So maybe use nameGenerator instead of generateName?

sindhurajain commented on src/main/java/duke/SubcommandParser.java:
```diff
@@ -0,0 +1,44 @@
+        List<Pair<String, String>> subcommandList = new ArrayList<>();
+        List<Pair<String, List<String>>> subcommandInternalList = new ArrayList<>();
+        String[] words = text.split(" ");
+        List<String> initial = new ArrayList<>();
```
Perhaps use the variable name to give more details as per the code quality guidelines (initial what?)

sindhurajain reviewed (commented):
Overall, good job. The code is readable and any violations in code quality/java coding standards are minor.

Darylgolden committed:
Start GUI

Darylgolden committed:
Implement basic GUI, echoing user

Darylgolden committed:
Switch to GUI

Darylgolden committed:
Update to use FXML

Darylgolden committed:
Update AI use

Darylgolden committed:
Merge branch 'Level-10'

Darylgolden committed:
Add Checkstyle configuration

Darylgolden committed:
Add checkstyle to gradle

Darylgolden committed:
Fix configuration

Darylgolden committed:
Merge branch 'A-CheckStyle'

Darylgolden committed:
Reformat code using IDEA

Darylgolden committed:
Fix more checkstyle issues

Darylgolden committed:
More minor code quality fixes

Darylgolden committed:
Add several assertions

Darylgolden committed:
Merge pull request #2 from Darylgolden/A-Assertions

Add several assertions

Darylgolden committed:
Merge pull request #3 from Darylgolden/A-CodeQuality

A code quality

Darylgolden committed:
Remove unused text redirection tests

Darylgolden committed:
Create CI with gradle

Darylgolden committed:
Merge pull request #4 from Darylgolden/A-CI

Create CI with gradle

Darylgolden committed:
Fix unit test data

Darylgolden committed:
Suppress Javadoc tests

Darylgolden committed:
Make attributes of ChatSession private

Darylgolden committed:
Make name attribute of task private

Darylgolden committed:
Suppress checkstyle warning for test

Darylgolden committed:
Fix trailing whitespace

Darylgolden committed:
Merge pull request #5 from Darylgolden/fix-tests

Fix tests

Darylgolden committed:
Restore functionality for exiting app with bye

Darylgolden committed:
Merge branch 'fix-tests'

Darylgolden committed:
Add Loan handling

This feature allows users to track their loans to people.
A Loan task type was added, as well as a LoanCommand
command to handle adding of loans.

Darylgolden committed:
Add functionality for saving and paying loans

Made changes to Loan to allow exporting data to the save file.
Made changes to Storage to allow parsing of saved loans.
Added the Pay command to allow users to track payment
of loans.

Darylgolden committed:
Ignore checkstyle false positive

Darylgolden committed:
Merge pull request #6 from Darylgolden/BCD-Extension

D-Loans

Darylgolden committed:
Change display picture to be circular

Darylgolden committed:
Add title window

Darylgolden committed:
Rename duke directories to goldbot

Darylgolden committed:
Remove unused imports

Darylgolden committed:
Rename remaining instances

Darylgolden committed:
wrap line

Darylgolden committed:
Fix trailing whitespace

Darylgolden committed:
Merge pull request #7 from Darylgolden/change-name

Rename duke directories to goldbot

Darylgolden committed:
Update documentation

Darylgolden committed:
Remove comment

Darylgolden committed:
Update product screenshot

Darylgolden committed:
Minor boolean variable change
